### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by Aduda-Boaz**
Add unit specs for all of the models' methods and validations.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+RSpec.describe User, type: :model do
+  subject { User.new(name: 'Philip', photo: 'https://...', bio: 'physician from Britain') }
+
+  before { subject.save }
+
+  it 'Validates the presence of Name' do
+    subject.name = nill
+    expect(subject).to_not be_valid
+  end
```
Comment (Gambit142) [inline spec/models/user_spec.rb:18]: - [x] Great work writing this validation test. I noticed that you did not write tests to confirm the presence of attributes for all models. Kindly write validation tests to confirm the presence of attributes for all models (`User`, `Post`, `Comment`, `Like`) created. For example:

```ruby on rails
describe 'validations' do
  it { should validate_presence_of(:name) }
end
```

File: app/models/post.rb [hunk 2]
```diff
@@ -7,6 +7,11 @@ def recent_comments
     comments.order(created_at: :desc).limit(6)
   end
 
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :comment_count, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
```
Comment (Gambit142) [inline app/models/post.rb]: - [x] Kindly write validation for the attribute `:like_count`. The `:like_count` must be an integer greater than or equal to zero. 👍

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  it 'Validates the length of Title to be less than or equal to 250' do
+    subject.title = 'a' * 251
+    expect(subject).to_not be_valid
+  end
+
+  it 'Validates increament of post counter after creation of post' do
+    user = User.create(name: 'Philip', photo: 'https://...', bio: 'physician from Britain')
+    expect(user.post_count).to eq(1)
+  end
+end
```
Comment (Gambit142) [inline spec/models/post_spec.rb]: - [x] One of the major project requirements is to write unit tests for all corresponding custom methods in each model. Kindly write a corresponding unit test for the `recent_comments` method in your `Post` model. 👍

File: spec/models/comment_spec.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  it 'Validates the presence of Post' do
+    @comment.post_id = nil
+    expect(@comment).to_not be_valid
+  end
+
+  it 'Validates the comment increament after creation' do
+    expect(@comment.post.comment_count).to eq(1)
+  end
```
Comment (Gambit142) [inline spec/models/comment_spec.rb:22]: - [x] [Optional] The `update_post_comments_count` method for which this unit test was written returns the wrong result. It should update the `comments_count` attribute in the `Post` model. 

```ruby on rails
 def update_post_comments_count
    post.increment!(:comments_count)
  end
```

File: spec/models/post_spec.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+  it 'Validates the length of Title to be less than or equal to 250' do
+    subject.title = 'a' * 251
+    expect(subject).to_not be_valid
+  end
+
+  it 'Validates increament of post counter after creation of post' do
+    user = User.create(name: 'Philip', photo: 'https://...', bio: 'physician from Britain')
+    expect(user.post_count).to eq(1)
+  end
```
Comment (Gambit142) [inline spec/models/post_spec.rb:26]: - [x] [Optional] The `update_post_count` method for which this unit test was written returns the wrong result. It should update the `posts_count` attribute in the `User` model. 

```ruby on rails
 def update_post_count
    user.increment!(:posts_count)
  end
```

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Aduda-Boaz 👋, 

Good job so far!

## Highlights
- `User`, `Post`, `Comment` and Like model have validations 💯 
- Wrote unit tests for `User`, `Post`, `Comment` and `Like` model 💯 

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ] The migration files that you have for this milestone are not complete. I noticed that in the migration files, all the columns for each table are missing. The foreign keys and indexes are also missing. Kindly update the migration files because I couldn't run your tests to check if all of them passed. 👍 

_Kindly check the comments under the review for other required changes._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

#### You can also consider the optional suggestions I made in the comments under the review. I recommend that you implement them because your application won't function properly if they are not implemented. 👍 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by aimemalaika — APPROVED** [review]
Hi @Aduda-Boaz , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

133591850-fdcb7dd3-dd7a-4274-81e7-0e59434148f7

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**